### PR TITLE
Account::create_and_fund_account()

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

So the executor-benchmark can use one less transaction for creating one test account.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

y

## Test Plan
smoke test

### CODE REVIEW

**Review by davidiw — DISMISSED** [review]
@movekevin , we should investigate how one might optionally create an account for certain coin transfers.

@msmouse , this API will be broken very, very soon and I don't think this should be a blocker for TestCoin -> Coin<TestCoin>. Your thoughts?

**Comment by msmouse** [comment]
@davidiw  how soon? I can resend it after the change.

I think `create_and_fund` is nice and explicit and unharmful vs `transfer_and_create_account_if_not_yet`.. but I'm glad to use that as well if provided.

**Comment by davidiw** [comment]
> @davidiw how soon? I can resend it after the change.
> 
> I think `create_and_fund` is nice and explicit and unharmful vs `transfer_and_create_account_if_not_yet`.. but I'm glad to use that as well if provided.

I mean, it is nuanced in that the Account shouldn't have a transfer function -- but we can follow along here: https://github.com/aptos-labs/aptos-core/pull/921

once that lands we should revisit, since I would really like to enable something like this... @movekevin

**Comment by movekevin** [comment]
> Account shouldn't have a t

Is the main use case for creating accounts during benchmarking (so not part of testnet/mainnet)?

**Comment by msmouse** [comment]
> Is the main use case for creating funded accounts during benchmarking (so not part of testnet/mainnet)?

yes, for me yes

File: aptos-move/framework/aptos-framework/sources/Account.move
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/Account.move:356]: This would now need to become Coin::transfer<TestCoin>(funder, account, amount).

An alternative here in case it's hard to find the appropriate funder is to mint an account some tokens after creation. You should be able to make a transaction (to create account and then mint) as the core resources account in the test.

Comment (msmouse) [reply aptos-move/framework/aptos-framework/sources/Account.move:356]: Yeah I can do a script transaction as well.

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/AccountUtils.move
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/AccountUtils.move:8]: This is pretty horrendous but I'll leave it to @movekevin to fix later :trollface:

**Review by davidiw — APPROVED** [review]
/land

File: aptos-move/framework/aptos-framework/sources/AccountUtils.move
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/AccountUtils.move:8]: Yeah I have an item on my TODO list for thinking about explicit vs implicit account creation vs having a utility function like this.

**Pull request closed without merging**
